src/scaffold/mod.rs

```diff
@@ -110,10 +110,28 @@ impl<'s> Scaffolder<'s> {
 
     /// Generates Solidity code from a `.tree` file.
     pub fn scaffold(&self, text: &str) -> crate::error::Result<String> {
-        let ast = syntax::parse(text)?;
+        let combiner = hir::combiner::Combiner::new();
```

Comment:
Same here, changes should be minimal: we split this function into more functions so that the pipeline described above is clearer to the reader.

Reply:
Sorry, so to confirm, here `Scaffolder::scaffold` is the main entrypoint which is being called via `Scaffold::run` - are you suggesting that we add a new `entrypoint` function (similar to the above) that is called instead? Something like this:
```
fn entrypoint() -> crate::error::Result<String> {
    // This is the &str read from the file.
    let trees = "TREES";
    let hirs = trees.split(TREE_SEPARATOR).map(|t| self.scaffold(t)?);
    Combiner::new().combine(&hirs);
    let pt = sol::Translator::new(self.solidity_version).translate(&hir);
    let source = sol::Formatter::new().emit(pt);
    let formatted = fmt(&source).expect("should format the emitted solidity code");
    Ok(formatted)
}

pub fn scaffold(&self, text: &str) -> anyhow::Result<Hir> {
    let ast = syntax::parse(text)?;
    let mut discoverer = modifiers::ModifierDiscoverer::new();
    let modifiers = discoverer.discover(&ast);
    let hir = hir::translator::Translator::new().translate(&ast, modifiers);
    Ok(hir)
}
```

Reply:
I should have been more clear. Something like:

```rust
    pub fn scaffold(&self, text: &str) -> crate::error::Result<String> {
        let hirs = text.split(TREE_SEPARATOR).map(|t| self.get_hir(t)?);
        Combiner::new().combine(&hirs);
        let pt = sol::Translator::new(self.solidity_version).translate(&hir);
        let source = sol::Formatter::new().emit(pt);
        let formatted = fmt(&source).expect("should format the emitted solidity code");

        Ok(formatted)
    }

    fn get_hir(&self, text: &str) -> anyhow::Result<hir::Hir> {
        let ast = syntax::parse(text)?;
        let mut discoverer = modifiers::ModifierDiscoverer::new();
        let modifiers = discoverer.discover(&ast);
        let hir = hir::translator::Translator::new().translate(&ast, modifiers);
        Ok(hir)
    }
```

Reply:
`get_hir` maybe isn't the best name, but wanted to give a better quickly

Reply:
As you can tell, `get_hir` is basically the same as `translate` in `src/hir/mod.rs`.

Reply:
Understood, thanks 👍